autovalet_system/src/autovalet_system.py

```diff
@@ -133,21 +134,28 @@ def registered_image_callback(self, color_msg, depth_msg):
         self.depth_img = self.bridge.imgmsg_to_cv2(depth_msg, desired_encoding="passthrough")
 
         self.depth_frame_id = depth_msg.header.frame_id
-        
+
         # if we're not in the PARK state AND the lane detector has been successfully initialized, detect the lane and publish
         if self.current_state != State.PARK and self.ld_init:
             # lane detection algo
             _, self.ego_line = self.laneDetector.detectLaneRGBD(self.color_img, self.depth_img)
 
         # self.processState()
-    
+
     def sendGoal(self):
         # generate goal from the egoline
         # if((rospy.get_time() - self.previous_time) > 2 and self.ego_line is not None):
         if self.ego_line is not None:
             self.current_goal = self.goalGenerator.generate_goal_from_egoline(self.ego_line, self.depth_frame_id)
             self.goal_pub.publish(self.current_goal)
             self.previous_time = rospy.get_time()
+            self.is_turning_left = False
+            return True
+        elif not self.is_turning_left and self.prev_state != State.START:
```

Comment:
trying to trace this -- what happens if we're ALREADY turning left and ego line is none

Reply:
I think then this fn returns false in that case?

Reply:
@SubramanianKrish prove this to me? wonder if this is the cause of the bug we were discussing

Reply:
I changed the logic - only when there are consecutive 5 frames w/o lines detected will a left goal be published.
That being said, if after a left goal is published, there are another 5 frames with no lines, another left goal will be published again.
lmk if the new code makes sense.